src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -52,24 +51,15 @@ public override Task<Unit> Handle(DidChangeTextDocumentParams request, Cancellat
 
             this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
 
-            if (IsBicepConfigFile(documentUri))
+            // Handle scenario where the bicepconfig.json file was opened prior to
+            // language service activation. If the config file was opened before the language server
+            // activation, there won't be an entry for it in the cache. We'll capture the state of the
+            // config file on disk when it's changes and cache it.
+            if (ConfigurationHelper.IsBicepConfigFile(documentUri) &&
```

Comment:
If we're needing to be thread-safe, this code shouldn't do a ContainsKey and then a TryAdd. It should use AddOrUpdate, your func will be passed the current value.

Reply:
@majastrz Just to verify...  Does this code need to be thread-safe?  In the ARM language server, on Taylor Mullen's suggestion I put a dispatcher that forced all requests from the host to be processed sequentially so we didn't need to worry about this.

Reply:
We haven't done that. So this does need to be thread safe.

Reply:
+1 on AddOrUpdate